src/DiscordBot.py

```diff
@@ -206,13 +207,42 @@ async def on_ready():
     """
     logger.info(f"Logged in as {bot.user.name}")
 
-    github_internship = GitHubUtilities(
-        token=GITHUB_TOKEN, repo_name="SimplifyJobs/Summer2025-Internships", isSummer=True, isCoop=True
+    try:
+        latest_repo = get_latest_internship_repo()
+        logger.info(f"Using latest internship repository: {latest_repo}")
+
+        github_internship = GitHubUtilities(
+            token=GITHUB_TOKEN, repo_name=f"SimplifyJobs/{latest_repo}", isSummer=True, isCoop=True
+        )
+        github_newgrad = GitHubUtilities(token=GITHUB_TOKEN, repo_name="SimplifyJobs/New-Grad-Positions")
+        job_utilities = JobsUtilities()
+        scheduled_task.start(github_internship, github_newgrad, job_utilities)  # Start the loop
+        health_check_task.start()  # Start health check task
+    except Exception as e:
+        logger.error(f"Failed to initialize bot with latest repository: {e}", exc_info=True)
+
+def get_latest_internship_repo():
+    g = Github(GITHUB_TOKEN)
+    org = g.get_organization("SimplifyJobs")
+    repos = org.get_repos()
+
+    matching_repos = []
+    for repo in repos:
+        if repo.name.startswith("Summer"):
+            suffix = repo.name[len("Summer"):]
+            year = suffix.split("-")[0]
+            if len(year) == 4 and year.isdigit():
+                matching_repos.append(repo.name)
+
+    if not matching_repos:
+        raise ValueError(
+            "No repositories matching the pattern 'SummerYYYY-Internships' were found in the organization SimplifyJobs. Make sure the naming format hasn't changed and that your GitHub token has the necessary permissions."
+        )
+
+    sorted_repos = sorted(
+        matching_repos, key=lambda name: int(name[len("Summer"):].split("-")[0]), reverse=True
```

Comment:
Why do you sort when you can just do max here?

Reply:
Good catch